### PR TITLE
Add an example, fuzzable Ruby example

### PR DESCRIPTION
**Pull request by robert**
r? @richo 

This takes the example Ruby harness from afl-ruby and makes sure we can fuzz it using roving.

### Test plan

Ran both example servers with a client, and we quickly found crashes in both the C and Ruby examples.

More comments inline

### CODE REVIEW

File: client/main.go [hunk 1]
```diff
@@ -73,11 +73,10 @@ func (f Fuzzer) run() error {
 }
 
 func (f AFLFuzzCommand) cmd() *exec.Cmd {
-	c := exec.Command(aflPath(),
+	fullCommandArgs := append([]string{
 		"-o", "output",
-		"-i", "input",
-		"-S",
-		"--", "./target")
+		"-i", "input"}, f.targetCommand...)
```
Comment (robert) [inline client/main.go]: `-S` was giving me some trouble, I'll make a task to come back and fix this stuff properly.

File: server/main.go [hunk 2]
```diff
@@ -112,11 +112,24 @@ func getState(c web.C, w http.ResponseWriter, r *http.Request) {
 	encoder.Encode(states)
 }
 
+// Returns info about the target.
+//
+// If a target is particularly large and will therefore take a long time
+// to download, we want to allow clients to supply their own copy of it.
+// In this case this endpoint will return the command that they should
+// run in order to fuzz it.
+func getTargetMeta(c web.C, w http.ResponseWriter, r *http.Request) {
```
Comment (robert) [inline server/main.go]: This new route is kind of the key to the whole thing, I imagine we might want to add extra stuff here in the future.

File: types/types.go [hunk 3]
```diff
@@ -42,6 +42,16 @@ type State struct {
 	Hangs   InputCorpus
 }
 
+type Target struct {
```
Comment (robert) [inline types/types.go]: We use these in both `client` and `server` so I thought putting them here made sense.

File: Makefile [hunk 4]
```diff
@@ -7,13 +7,16 @@ client/client: $(wildcard client/*.go) $(wildcard types/*.go)
 	cd client && go build
 
 example-target: check-env
-	AFL_HARDEN=1 $(AFL)/afl-clang -o example/server/$@ example/server/target.c
+	AFL_HARDEN=1 $(AFL)/afl-clang -o examples/server/$@ examples/server/target.c
 
-example-server: server/server
-	$(CURDIR)/server/server $(CURDIR)/example/server
+example-server-c: server/server
```
Comment (richo) [inline Makefile]: These should both be marked `.PHONY` at the bottom since it doesn't produce a concrete artifact

**Comment by richo** [comment]
Nothing sticks out as a bad idea (I'm assuming you tested it end to end)

r+

**Pull request merged**